diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullableTypeRule.kt

```diff
@@ -0,0 +1,151 @@
+                "listOf", "mapOf", "arrayOf", "sequenceOf", "setOf")
+    }
+
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        isFixMode = autoCorrect
+        emitWarn = emit
+
+        if (node.elementType == PROPERTY)
+            checkProperty(node)
+
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun checkProperty(node: ASTNode) {
+        if (node.hasChildOfType(VAL_KEYWORD) && node.hasChildOfType(EQ) && node.hasChildOfType(TYPE_REFERENCE)) {
+            val typeReferenceNode = node.findChildByType(TYPE_REFERENCE)!!
+            // check that property has nullable type, right value one of allow expression
+            if (!node.hasChildOfType(NULL) &&
+                    !node.hasChildOfType(DOT_QUALIFIED_EXPRESSION) &&
+                    typeReferenceNode.hasChildOfType(NULLABLE_TYPE) &&
+                    typeReferenceNode.findChildByType(NULLABLE_TYPE)!!.hasChildOfType(QUEST) &&
+                    (node.findChildByType(CALL_EXPRESSION)?.findChildByType(REFERENCE_EXPRESSION) == null ||
+                            node.findChildByType(CALL_EXPRESSION)!!.findChildByType(REFERENCE_EXPRESSION)!!.text in allowExpression)) {
+                    NULLABLE_PROPERTY_TYPE.warn(configRules, emitWarn, isFixMode, "don't use nullable type",
+                            node.findChildByType(TYPE_REFERENCE)!!.startOffset, node)
+            } else if (node.hasChildOfType(NULL)) {
+                val fixedParam = findFixableParam(node)
+                NULLABLE_PROPERTY_TYPE.warnAndFix(configRules, emitWarn, isFixMode, "initialize explicitly",
+                        node.findChildByType(NULL)!!.startOffset, node, fixedParam != null) {
+                    if (fixedParam != null) findSubstitution(node, fixedParam)
+                }
+            }
+        }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun findFixableParam(node: ASTNode): FixedParam? {
+        val reference = node.findChildByType(TYPE_REFERENCE)!!.findChildByType(NULLABLE_TYPE)!!.findChildByType(USER_TYPE)?.findChildByType(REFERENCE_EXPRESSION)
+                ?: return null
+        return when (reference.text) {
+            "Boolean" -> FixedParam(BOOLEAN_CONSTANT, TRUE_KEYWORD, "true")
+            "Int", "Short", "Byte" -> FixedParam(INTEGER_CONSTANT, INTEGER_LITERAL, "0")
+            "Double" -> FixedParam(FLOAT_CONSTANT, FLOAT_LITERAL, "0.0")
+            "Float" -> FixedParam(FLOAT_CONSTANT, FLOAT_LITERAL, "0.0F")
+            "Long" -> FixedParam(INTEGER_CONSTANT, INTEGER_LITERAL, "0L")
+            "Char" -> FixedParam(CHARACTER_CONSTANT, CHARACTER_LITERAL, "")
```

Comment:
Does this mean you will try to assign empty string into a `Char` property? You could use something like `Char.MIN_VALUE` here instead.